packages/core/src/Nodes/FunctionNode.ts

```diff
@@ -28,71 +32,94 @@ export class FunctionNode extends Node {
-  }) {
-    const alpha = 'abcdefghijklmnop';
-    const inSocketKeys: string[] = [];
-    const inSocketSpecs: { [name: string]: string } = {};
+const alpha = 'abcdefghijklmnop';
+const getAlphabeticalKey = (index: number) => alpha[index];
 
-    const in2 = props.in ?? [];
-    in2.forEach((inElement, index) => {
-      if (typeof inElement === 'string') {
-        inSocketSpecs[alpha[index]] = inElement;
-        inSocketKeys.push(alpha[index]);
-      } else {
-        const inElementKeys = Object.keys(inElement);
-        Assert.mustEqual(inElementKeys.length, 1);
-        inSocketSpecs[inElementKeys[0]] = inElement[inElementKeys[0]];
-        inSocketKeys.push(inElementKeys[0]);
-      }
-    });
-
-    const outSocketSpecs: { [name: string]: string } = {};
-    const outSocketKeys: string[] = [];
+type OrderedSocketsResult = {
+  socketKeys: string[];
+  sockets: SocketsDefinition;
+};
+/** Converts list of sockets specifying value type names to an ordeered list of sockets,
+ */
+const toOrderedSockets = (
+  socketValueTypes: string[] | string | undefined = [],
+  getKey: (index: number) => string
+): OrderedSocketsResult => {
+  // const alpha = 'abcdefghijklmnop';
 
-    const out2 =
-      typeof props.out === 'string' ? [{ result: props.out }] : props.out;
-    out2.forEach((outElement) => {
-      const outElementKeys = Object.keys(outElement);
-      Assert.mustEqual(outElementKeys.length, 1);
-      outSocketSpecs[outElementKeys[0]] = outElement[outElementKeys[0]];
-      outSocketKeys.push(outElementKeys[0]);
-    });
+  return [...socketValueTypes].reduce(
+    ({ socketKeys, sockets }: OrderedSocketsResult, socketValueType) => {
+      const name = getKey(socketKeys.length);
+      return {
+        socketKeys: [...sockets.socketsKeys, name],
+        sockets: {
+          ...sockets,
+          [name]: socketValueType
+        }
+      };
+    },
+    {
+      socketKeys: [],
+      sockets: {}
+    }
+  );
+};
 
-    super(
-      props.name,
-      'Logic',
-      props.label ?? props.name,
-      (description, graph) => {
-        return new FunctionNode(
-          description,
-          graph,
-          inSocketKeys.map((inKey) => new Socket(inSocketSpecs[inKey], inKey)),
-          outSocketKeys.map(
-            (outKey) => new Socket(outSocketSpecs[outKey], outKey)
-          ),
-          (node) => {
-            const args = inSocketKeys.map((inKey) => node.readInput(inKey));
-            const results = props.exec(...args);
-            if (outSocketKeys.length === 1 && outSocketKeys[0] === 'result') {
-              node.writeOutput('result', results);
-            } else {
-              outSocketKeys.forEach((outKey) => {
-                node.writeOutput(outKey, results[outKey]);
-              });
-            }
-          }
-        );
-      },
-      props.aliases
+export function makeInNOutFunctionDesc({
+  in: inputValueTypes,
+  inputKeys,
+  out,
+  exec,
+  ...rest
+}: {
```

Comment:
I'm not sure why the `helpDescription` prop was removed when creating the `FunctionDesc` class. I think we should include it again as an otional prop for every makeFunctionDesc method as it's pretty useful for docs and visual editor

Reply:
An example of using `helpDescription` on a patched FunctionDesc (as mentioned, the implementation doesn't accept that prop right now. As you can see, the docs are much better with it, we can even include links for references. If the node doesn't have the `helpDescription` field defined, it will default to the `label` which is not really useful
![image](https://user-images.githubusercontent.com/5903438/209564952-3f2625bb-7c73-4db9-b071-8d95109f8c3e.png)
![image](https://user-images.githubusercontent.com/5903438/209564977-855031d8-1213-427f-be97-882a9ab22f3f.png)
![image](https://user-images.githubusercontent.com/5903438/209565010-bfef844f-2596-4f0b-83f1-f7c0d3bf68a1.png)

Reply:
Thanks for pointing that out, its removal was unintentional; make sure to include it.